tests/tensorboard_test.py

```diff
@@ -0,0 +1,204 @@
+    return event_value_list[0]['value']
+
+  def assert_encoded_audio_files_equal(self, actual_audio, expected_audio):
+    # expected_audio is trimmed to -1.0 to 1.0
+    self.assertFalse(onp.allclose(actual_audio.numpy(), expected_audio.numpy()))
+
+    # trim the audio to assert the values.
+    trimmed_audio = onp.clip(onp.array(actual_audio), -1, 1)
+    self.assertTrue(
+        onp.allclose(trimmed_audio, expected_audio.numpy(), atol=1e-04))
+
+  def test_summarywriter_flush_after_close(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    summary_writer.close()
+    with self.assertRaises(AttributeError):
+      summary_writer.flush()
+
+  def test_summarywriter_scalar(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    # Write the scalar and check if the event exists and check data.
+    float_value = 99.1232
+    summary_writer.scalar(tag='scalar_test', value=float_value, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'scalar_test')
+    self.assertTrue(onp.allclose(
+        tensor_util.make_ndarray(summary_value.tensor).item(),
+        float_value))
+
+  def test_summarywriter_text(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    text = 'hello world.'
+    summary_writer.text(tag='text_test', textdata=text, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'text_test')
+    self.assertEqual(
+        tensor_util.make_ndarray(summary_value.tensor).item().decode('utf-8'),
+        text)
+
+  def test_summarywriter_image(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = tf.random.uniform(shape=[30, 30, 3])
+    img = tf.cast(255 * img, tf.uint8)
+    summary_writer.image(tag='image_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, 'image_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    self.assertTrue(onp.allclose(img, expected_img.numpy()))
+
+  def test_summarywriter_2dimage_scaled(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = tf.random.uniform(shape=[30, 30])
+    img = tf.cast(255 * img, tf.uint8)
+    summary_writer.image(tag='2dimage_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, '2dimage_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    # assert the image was increased in dimension
+    self.assertEqual(expected_img.shape, (30, 30, 3))
+
+  def test_summarywriter_single_channel_image_scaled(self):
+    log_dir = tempfile.mkdtemp()
```

Comment:
Seems like you could move `log_dir` and `summary_writer` creation into a setUp method.

Reply:
Yeah I had something like that initially. Just wasn't sure if it actually adds a lot of value v/s being explicit and verbose about how the test is being setup (temp file + summary writer). Felt the latter was more readable.. WDTY?

Reply:
Personal preference would be to move the boilerplate out of the test methods, but don't have a strong opinion. Please go with what you prefer / have already implemented.